test/test_source.cpp

```diff
@@ -0,0 +1,187 @@
+                    "}", headersNew1};
+
+                dim3 gridNew1(1);
+                dim3 blockNew1(5);
+                
+                sourceNew1.program("cuda_add_with_header").compile().configure(gridNew1, blockNew1).launch(args);
+                
+                //B3. Comparing the results
+                for (int i=0;i<5;i++) REQUIRE(hOut[i] == hostCompareOutput_1[i]);
+            }    
+        }
+
+        //C1. Checking for consistencies for running various functions in a kernel-code
+        WHEN("More one functions in a kernel-code are used."){
+            args.clear();
+            args.emplace_back(KernelArg{hX, bufferSize});
+            args.emplace_back(KernelArg{hY, bufferSize});
+            args.emplace_back(KernelArg{hOut_Multiply1, bufferSize, true});
+            args.emplace_back(KernelArg(&datasize));
+            
+            Headers headersNew2;
+            headersNew2.insert(Header{"cuda_runtime.h"});
+
+            Source sourceNew2{
+                "#include \"cuda_runtime.h\"\n"
+                "extern \"C\"\n"
+                "__global__ void cuda_add_normal(int *x, int *y, int *out, int "
+                "datasize) {\n"
+                " int i = threadIdx.x;\n"
+                " out[i] = x[i] + y[i];\n"
+                "}\n"
+                "extern \"C\"\n"
+                "__global__ void cuda_add_five(int *x, int *y, int *out, int"
+                " datasize) {\n"
+                " int i = threadIdx.x;\n"
+                " out[i] = x[i] + y[i] + 5;\n"
+                "}\n", headersNew2};
+
+            /*C2. Compilation of Kernels through the creation of a 
+            program from kernel - source*/
+            THEN("The results are consistent with the given controlled output."){
+                dim3 gridNew2(1);
+                dim3 blockNew2(5);
+                
+                sourceNew2.program("cuda_add_five").compile().configure(gridNew2, blockNew2).launch(args);
+                
+                //C3. Compiling another function in a kernel-code.
+                args.clear();
+                args.emplace_back(KernelArg{hX, bufferSize});
+                args.emplace_back(KernelArg{hY, bufferSize});
+                args.emplace_back(KernelArg{hOut_Multiply2, bufferSize, true});
+                args.emplace_back(KernelArg(&datasize));   
+               
+                sourceNew2.program("cuda_add_normal").compile().configure(gridNew2, blockNew2).launch(args);
+
+                //C4. Comparing the results for the use of multiple functions in a kernel-code
+                for (int i=0;i<5;i++) REQUIRE(hOut_Multiply1[i] == hostCompareOutput_1[i]);
+                for (int i=0;i<5;i++) REQUIRE(hOut_Multiply2[i] == hostCompareOutput_Controlled[i]);
+            }    
+        }
+
+        //D1. Checking for the consistencies of declaring functions
+        WHEN("Naming a kernel-compilation function is consistent to that of that function of a kernel-code"){
```

Comment:
I don't really understand the purpose of this? the mangled function name that is being executed is not being compared here.

Reply:
Das Ziel ist zu überprüfen, ob die Namengebung vom Benutzer auf der Kompilation - Seite des Kernels genau der von der Funktion des Kernel-Codes entspricht.